layers/+spacemacs/spacemacs-defaults/funcs.el

```diff
@@ -479,15 +479,61 @@ If the universal prefix argument is used then will the windows too."
             (if dedicated "no longer " "")
             (buffer-name))))
 
-;; http://camdez.com/blog/2013/11/14/emacs-show-buffer-file-name/
-(defun spacemacs/show-and-copy-buffer-filename ()
-  "Show and copy the full path to the current file in the minibuffer."
+
+;; Copy file path
+(defun spacemacs/copy-directory-path ()
+  "Copy and show the directory path of the current buffer.
+If the buffer is not visiting a file, use the `list-buffers-directory'
+variable as a fallback to display the directory, useful in buffers like the
+ones created by `magit' and `dired'."
+  (interactive)
+  (if-let (directory-name (or
+                            (if-let (file-name (buffer-file-name))
+                              (file-name-directory file-name))
+                            list-buffers-directory))
```

Comment:
The `or` seems unnecessary.  How about:

```emacs-lisp
(if-let (directory-name (if-let (file-name (buffer-file-name))
￼                            (file-name-directory file-name)
                          list-buffers-directory))
```

Reply:
I'm still wrapping my head around using `if-let` properly. I wish I had it in other languages as well, seems like it compacts a lot of branching code.